### PR TITLE
Adding RPC endpoint support (Client.fetch_application)

### PR DESCRIPTION
**Pull request by lol219**
## Summary

This PR adds a fetch_application() method to client.py and a matching get_application() in http.py. 

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

File: discord/http.py [hunk 1]
```diff
@@ -811,7 +811,8 @@ def pin_message(self, channel_id: Snowflake, message_id: Snowflake, reason: Opti
             message_id=message_id,
         )
         return self.request(r, reason=reason)
-
+     def get_application(self, application_id: Snowflake, /) -> Response[appinfo.PartialAppInfo]:
+        return self.request(Route('GET', '/applications/{application_id}/rpc', application_id=application_id))
```
Comment (Lulalaby) [inline discord/http.py:815]:
```suggestion

     def get_application(self, application_id: Snowflake, /) -> Response[appinfo.PartialAppInfo]:
        return self.request(Route('GET', '/applications/{application_id}/rpc', application_id=application_id))

```

**Review by lol219 — COMMENTED** [review]
yes

File: discord/flags.py [hunk 2]
```diff
@@ -282,6 +282,15 @@ def join_notification_replies(self):
...
 
+    @flag_value
+    def app_commands_badge(self):
+        """:class:`bool`: Returns ``True`` if the application has registered a global application
+        command. This shows up as a badge in the bot's profile.
+
+         .. versionadded:: 2.1
+        """
+        return 1 << 23
+
```
Comment (Lulalaby) [inline discord/flags.py:293]:
```suggestion
```

**Review by Lulalaby — APPROVED** [review]

**Comment by lol219** [comment]
thank you for fixing these errors lol i was about to update them

**Review by Lulalaby — DISMISSED** [review]

File: discord/client.py [hunk 3]
```diff
@@ -482,7 +482,6 @@ async def before_identify_hook(self, shard_id: Optional[int], *, initial: bool =
             await asyncio.sleep(5.0)
 
     # login state management
```
Comment (Lulalaby) [inline discord/client.py:484]:
```suggestion
    # login state management

```

**Review by Lulalaby — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Comment by searchify7** [comment]
Yep also we need to make a code example of how to use it, maybe?

**Comment by lol219** [comment]
yes , why not .

**Comment by Dorukyum** [comment]
> Yep also we need to make a code example of how to use it, maybe?

This is only a function, the docs should be enough.

**Comment by searchify7** [comment]
> > Yep also we need to make a code example of how to use it, maybe?
> 
> This is only a function, the docs should be enough.

Yeah you're right but other people can't understand how to use it

**Review by lol219 — COMMENTED** [review]
Uh I think we forgot version added in documentation @Lulalaby @Dorukyum

**Comment by lol219** [comment]
Uh I think we forgot version added in documentation @Lulalaby @Dorukyum

**Pull request merged**